build/buildRunTimeImageBases.sh

```diff
@@ -123,24 +124,48 @@ for dockerFile in $dockerFiles; do
     platformName="${PARTS[0]}"
     platformVersion="${PARTS[1]}"
 
-    # Set $localImageTagName to the following format: oryxdevmcr.azurecr.io/public/oryx/base:{platformName}-{platformVersion}
-    localImageTagName="$BASE_IMAGES_REPO:$platformName-$platformVersion"
+    if shouldStageRuntimeVersion $platformName $platformVersion ; then
+        # skip the normal base.{ostype}.Dockerfile if this version should be staged
+        if [[ "$dockerFile" != *"staging"* ]]; then
+            continue
+        fi
+        # Set $localImageTagName to the following format: oryxdevmcr.azurecr.io/staging/oryx/base:{platformName}-{platformVersion}
+        localImageTagName="$BASE_IMAGES_STAGING_REPO:$platformName-$platformVersion"
+    else
+        # Set $localImageTagName to the following format: oryxdevmcr.azurecr.io/public/oryx/base:{platformName}-{platformVersion}
+        localImageTagName="$BASE_IMAGES_REPO:$platformName-$platformVersion"
+    fi
 
     echo
     echo "Building image '$localImageTagName' for Dockerfile located at '$dockerFile'..."
 
     cd $REPO_DIR
-    echo
 
-    docker build -f $dockerFile \
-        -t $localImageTagName \
-        --build-arg CACHEBUST=$(date +%s) \
-        --build-arg NODE14_VERSION=$NODE14_VERSION \
-        --build-arg NODE16_VERSION=$NODE16_VERSION \
-        --build-arg NODE18_VERSION=$NODE18_VERSION \
-        --build-arg DEBIAN_FLAVOR=$runtimeImageDebianFlavor \
-        $labels \
-        .
+    echo
+    if shouldStageRuntimeVersion $platformName $platformVersion ; then
+        # pass in env var as a secret, which is mounted during a single run command of the build
+        # https://github.com/docker/buildx/blob/master/docs/reference/buildx_build.md#secret
+        DOCKER_BUILDKIT=1 docker build -f $dockerFile \
+            -t $localImageTagName \
+            --build-arg CACHEBUST=$(date +%s) \
+            --build-arg NODE14_VERSION=$NODE14_VERSION \
+            --build-arg NODE16_VERSION=$NODE16_VERSION \
+            --build-arg NODE18_VERSION=$NODE18_VERSION \
+            --build-arg DEBIAN_FLAVOR=$runtimeImageDebianFlavor \
+            --secret id=DOTNET_PRIVATE_STORAGE_ACCOUNT_ACCESS_TOKEN,env=DOTNET_PRIVATE_STORAGE_ACCOUNT_ACCESS_TOKEN \
```

Comment:
Just to be sure: is this the actual access token, or is this variable corresponding to the ID of the secret where the access token is stored? I assume this is the ID of the secret, but then it could be a good thing to add an _ID suffix to the env variable or something?

Reply:
see: https://github.com/microsoft/Oryx/pull/1643/commits/c290aaaec36180ae5710cc42294b6672703e82db

Reply:
the `env=DOTNET_PRIVATE_STORAGE_ACCOUNT_ACCESS_TOKEN` tells the code to look at the DOTNET_PRIVATE_STORAGE_ACCOUNT_ACCESS_TOKEN env var where this was run, and use that as the secret

Reply:
Thanks Paul!